### ISSUE TITLE
Using error handling pages causes multiple identities to be populated in HttpContext.User.Identities

### ISSUE THREAD

**Issue by jeffputz** [aspnet/Identity#1306]
For back story, check out a bug a user found in my project:
https://github.com/POPWorldMedia/POPForums/issues/54

The TL;DR version is that configuring error handling pages in startup, specifically `app.UseStatusCodePagesWithReExecute(..)` and `app.UseExceptionHandler(..)`, appears to cause HttpContext.User.Identities to be populated with two copies of any configured identities when the framework pushes you over to the new pretty error page.

In my case, this fails in two ways, because I have an action filter that manipulates the identity, calling it by name from the collection using `SingleOrDefault` on the `Identities` collection (line 49):
https://github.com/POPWorldMedia/POPForums/blob/master/src/PopForums.Mvc/Areas/Forums/Authorization/PopForumsUserAttribute.cs

Then it also fails because I attempt to get the identity in the same way to get user data before recording the error (line 37):
https://github.com/POPWorldMedia/POPForums/blob/master/src/PopForums.Mvc/Areas/Forums/Extensions/Logger.cs

While I realize that I could avoid the error by simply getting `FirstOrDefault`, having more than one copy of the identity seems not right if for some reason you need to rely on one authoritative copy.

My auth setup looks like this (some of this is plumbing to decide which providers to setup):
`

			var setupService = app.ApplicationServices.GetService<ISetupService>();
			if (!setupService.IsConnectionPossible() || !setupService.IsDatabaseSetup())
				return app;
			app.UseCookieAuthentication(new CookieAuthenticationOptions
			{
				AuthenticationScheme = PopForumsAuthorizationDefaults.AuthenticationScheme,
				CookieName = PopForumsAuthorizationDefaults.CookieName,
				AutomaticAuthenticate = true
			});
			var settingsManager = app.ApplicationServices.GetService<ISettingsManager>();
			var settings = settingsManager.Current;

			app.UseCookieAuthentication(new CookieAuthenticationOptions
			{
				AuthenticationScheme = ExternalUserAssociationManager.AuthenticationContextName,
				CookieName = ExternalUserAssociationManager.AuthenticationContextName,
				ExpireTimeSpan = TimeSpan.FromMinutes(5)
			});

			if (settings.UseTwitterLogin)
				app.UseTwitterAuthentication(new TwitterOptions
				{
					ConsumerKey = settings.TwitterConsumerKey,
					ConsumerSecret = settings.TwitterConsumerSecret
				});

			if (settings.UseMicrosoftLogin)
				app.UseMicrosoftAccountAuthentication(new MicrosoftAccountOptions
				{
					ClientId = settings.MicrosoftClientID,
					ClientSecret = settings.MicrosoftClientSecret
				});

			if (settings.UseFacebookLogin)
				app.UseFacebookAuthentication(new FacebookOptions
				{
					AppId = settings.FacebookAppID,
					AppSecret = settings.FacebookAppSecret
				});

			if (settings.UseGoogleLogin)
				app.UseGoogleAuthentication(new GoogleOptions
				{
					ClientId = settings.GoogleClientId,
					ClientSecret = settings.GoogleClientSecret
				});

			return app;
`

**Comment by blowdart** [comment]
As there's no Identity here, I'm moving this to security.
